.github/workflows/0-everything.yml

```diff
@@ -0,0 +1,93 @@
+# ----------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+#
+# THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
+# OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+# ----------------------------------------------------------------------------------
+
+name: 0 - Everything
+
+on:
+  workflow_dispatch:
+    inputs:
+      hubNetworkType:
+        type: choice
+        description: "Select the hub network type"
+        required: true
+        options:
+          - "HubNetworkWithNVA"
+          - "HubNetworkWithAzureFirewall"
+        default: "HubNetworkWithAzureFirewall"
+      subscriptionIds:
+        type: string
+        description: The subscription ID(s) to use for the deployment.
+        required: true
+
+defaults:
+  run:
+    shell: pwsh
+    working-directory: scripts/deployments
+
+jobs:
+  everything:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Required modules
+      run: |
+        Install-Module Az -Force
+        Install-Module powershell-yaml -Force
+
+    - name: Deploy Management Groups
+      run: |
+        ./RunWorkflows.ps1 `
+          -DeployManagementGroups `
+          -LoginServicePrincipalJson '${{secrets.AZURE_CREDENTIALS}}' `
+          -GitHubRepo ${env:GITHUB_REPOSITORY} `
+          -GitHubRef ${env:GITHUB_REF}
+
+    - name: Deploy Custom Roles
+      run: |
+        ./RunWorkflows.ps1 `
+          -DeployRoles `
+          -LoginServicePrincipalJson '${{secrets.AZURE_CREDENTIALS}}' `
+          -GitHubRepo ${env:GITHUB_REPOSITORY} `
+          -GitHubRef ${env:GITHUB_REF}
+
+    - name: Deploy Logging
+      run: |
+        ./RunWorkflows.ps1 `
+          -DeployLogging `
+          -LoginServicePrincipalJson '${{secrets.AZURE_CREDENTIALS}}' `
+          -GitHubRepo ${env:GITHUB_REPOSITORY} `
+          -GitHubRef ${env:GITHUB_REF}
+
+    - name: Deploy Policy
+      run: |
+        ./RunWorkflows.ps1 `
+          -DeployPolicy `
+          -LoginServicePrincipalJson '${{secrets.AZURE_CREDENTIALS}}' `
+          -GitHubRepo ${env:GITHUB_REPOSITORY} `
+          -GitHubRef ${env:GITHUB_REF}
+
+    - name: Deploy ${{github.event.inputs.hubNetworkType}}
```

Comment:
I think we should breakout this step into 2

- With NVA (that also takes in username and password from GitHub secrets)
- With Azure Firewall (without the username & password)

This can be conditionally run with `if: github.event.inputs.hubNetworkType == 'HubNetworkWithNVA'` and `if: github.event.inputs.hubNetworkType == 'HubNetworkWithAzureFirewall'`

This approach will also allow us to put in a user friendly name similar to the other tasks such as "Deploy Hub Networking with Azure Firewall"

Reply:
I think it's clear the original way with neither approach being significantly better than the other. The original approach has the benefit of being better/less code (more readable) and does not suffer the side-effect of modified approach showing both hub network steps with one as skipped. However, I will change it per your suggestion as either approach is fine with me.

![image](https://user-images.githubusercontent.com/755941/167309074-fc2f47f1-737a-4cd4-a82c-b07aadbd166a.png)